src/Agent.Worker/JobExtension.cs

```diff
@@ -118,6 +118,31 @@ public async Task<List<IStep>> InitializeJob(IExecutionContext jobContext, Pipel
 
                         taskConditionMap[task.Id] = condition;
                     }
+                    foreach (var knob in AgentKnobs.GetAllKnobs())
+                    {
+                        var value = knob.GetValue(jobContext);
+                        if (value.Source != KnobSource.BuiltInDefault)
+                        {
+                            var tag = "";
+                            if (knob.IsDeprecated)
```

Comment:
I might prefer a deprecation message to a bool here. That way we can add context on why/potential removal date/workarounds

Reply:
Would it make sense to update the Description when a knob is deprecated and keep IsDeprecated as boolean?

Reply:
I'm fine with that